logger/logger.go

```diff
@@ -270,8 +270,15 @@ func (logger Logger) makeRequest(logmsg Message) {
 	if err != nil {
 		log.Fatalln(err)
 	}
+	
+	req, err := http.NewRequest("POST", options.IngestURL, bytes.NewBuffer(bytesRepresentation))
+	req.Header.Set("user-agent", "go/1.0.0")
```

Comment:
@dm36, I know versioning is kinda weird for this package but hardcoding it within the `logger.go` isn't good, I think - you need to figure out a way for storing the version separately and call it here.

Reply:
I think based on our namespace conversation, `go` isn't descriptive enough for the user agent value.

Reply:
My impression was the namespace conversation was around package and repository naming- not the user-agent @darinspivey. What would be an apt name for the user-agent- "logger-go/1.0.0"?

Reply:
Ah, well, it's both.  The user agent value comes from the package name and version (which includes the namespace).  In the case of Go, I'm not sure if it has a namespace.  We want `logdna` somewhere in that (so for `npm`, it's @logdna/xxx), so perhaps it's just `logdna/logger-go/1.0.0`?

Reply:
just noting that ingester does some validation on these agent/client useragents, something like `logdna/logger-go/1.0.0` would not (currently) satisfy that validation - `logdna-logger-go/1.0.0` would though.

Reply:
Hrm, good point, in that case I'd lean more towards `logger-go/1.0.0` then because we've been trying to not put `logdna` in proper names (like repos, etc).

Reply:
@mdeltito, we will need to update and revise that logic on the ingester side since we are making lots of changes into what gets sent thru `user-agent` - I think we can work on server-side once we have new and proper list of the changes.

Reply:
@mdeltito, @darinspivey, @dm36, according to the current setup / checks, only `go/<version>` is gonna work - I'm fine with the naming for now since it is internal-facing and we can change it later once we make changes to the ingester. I'm mainly wondering whether keeping the version and/or versioning inside the main implementation is right or wrong. For example, in the LogSpout, I keep it inside `Dockerfile`: https://github.com/logdna/logspout/blob/master/Dockerfile#L2 and use it like this: https://github.com/logdna/logspout/blob/master/logdna/adapter/adapter.go#L198 - in that case, `Dockerfile` plays the same role as `package.json` which is for `npm`

Reply:
@smusali yes there was a recent addition to ingester to support `go`, which will need to change per the outcome here.

I agree we shouldn't have the current package version embedded inline here, I think we'd want to apply the version dynamically at the time of build/deployment based off the tag 

https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications